jax/api.py

```diff
@@ -1680,8 +1684,8 @@ def _jvp(fun: lu.WrappedFun, primals, tangents):
            "tree structure {}")
     raise TypeError(msg.format(tree_def, tree_def_2))
   for p, t in safe_zip(ps_flat, ts_flat):
-    if _dtype(p) != _dtype(t):
-      msg = ("primal and tangent arguments to jax.jvp must have equal types; "
+    if core.primal_dtype_to_tangent_dtype(_dtype(p)) != _dtype(t):
+      msg = ("primal and tangent arguments to jax.jvp do not match; must be equal or float0 in case of int/bool. "
```

Comment:
```suggestion
msg = ("primal and tangent arguments to jax.jvp do not match; "
       "dtypes must be equal, or in case of int/bool primal dtype "
       "the tangent dtype must be float0."
       f"Got primal dtype {_dtype(p)} and so expected tangent dtype "
       f"{core.primal_dtype_to_tangent_dtype(_dtype(p))}, but got "
       f"tangent dtype {_dtype(t)} instead.")
```